### PR TITLE
8 binary and relational operators

### PR DESCRIPTION
**Pull request by Cdebus**
Concerns #8 
Elementwise relational (eq, ne, le, lt, ge, gt) and binary (add, sub, div, mul, pow) opeartions on tensors
Includes:
- moving of operation implementations to operations.py
- tunneling of operations in tensor.py to operations
- testing for binary and relational operations
- implementation of equal() to return boolean for equality (values, size) of two tensors
- documentation of operations

### CODE REVIEW

File: heat/core/tests/test_operations.py [hunk 1]
```diff
@@ -2,9 +2,28 @@
 import torch
 
 import heat as ht
+import numpy as np
```
Comment (Markus-Goetz) [inline heat/core/tests/test_operations.py]: numpy should go with the third-party packages up top before importing heat

File: heat/core/operations.py [hunk 2]
```diff
@@ -23,7 +25,19 @@
     'sum',
     'transpose',
     'tril',
-    'triu'
+    'triu',
```
[the remark this answers is not in the document]
Comment (Cdebus) [reply heat/core/operations.py]: Sorry, wasn't aware that this is convention. Will fix it

**Comment by Markus-Goetz** [comment]
Due to Daniel's nice work on subdividing the functions into separate files, you might need merge the master once again into your branch and then move +, -, /, * etc. into arithmetics.py and eq, neq, etc. into relational.py. Sorry about the sudden change. At the moment the code organization can at times still change significantly.

File: heat/core/operations.py [hunk 3]
```diff
@@ -693,6 +707,326 @@ def triu(m, k=0):
...
+      The first operand involved in the addition
+
+      t2: tensor or scalar
+      The second operand involved in the addition
+
+
+      Returns
+      -------
+      result: ht.tensor
+      A tensor containing the results of element-wise addition of t1 and t2.
```
[the remark this answers is not in the document]
Comment (Cdebus) [reply heat/core/operations.py]: What do you mean by "distinction between the tensor method and the module-level function call" ?
Examples for both ht.add(T1,T2) and T1.__add__(T2) ?

[the remark this answers is not in the document]
Comment (Markus-Goetz) [reply heat/core/operations.py]: yes exactly

**Comment by Cdebus** [comment]
I have amended the branch to include all requested changes, merged the new master after refactorization and distributed the operations accordingly. For relational operations, there is a new module relations.py .
Please recheck that everything is in order and i have not forgotten anything

File: heat/core/tests/test_operations.py (hunk 1 above)
Comment (Cdebus) [reply heat/core/tests/test_operations.py]: numpy was only used for some internal tests, it isn't needed anymore, so i removed it from there

**Comment by Markus-Goetz** [comment]
Nice work!

**Comment by rainman110** [comment]
Great work Charlie :tada:

**Pull request merged**